src/scaling/scaling.jl

```diff
@@ -0,0 +1,53 @@
+type ScaledInterpolation{T,N,ITPT,IT,GT,RT} <: AbstractInterpolationWrapper{T,N,ITPT,IT,GT}
+    itp::ITPT
+    ranges::RT
+end
+ScaledInterpolation{T,ITPT,IT,GT,RT}(::Type{T}, N, itp::ITPT, ::Type{IT}, ::Type{GT}, ranges::RT) =
+    ScaledInterpolation{T,N,ITPT,IT,GT,RT}(itp, ranges)
+function scale{T,N,IT,GT}(itp::AbstractInterpolation{T,N,IT,GT}, ranges::Range...)
+    length(ranges) == N || error("Must scale $N-dimensional interpolation object with exactly $N ranges (you used $(length(ranges)))")
+    for d in 1:N
+        iextract(IT,d) != NoInterp || ranges[d] == 1:size(itp,d) || error("NoInterp dimension $d must be scaled with unit range 1:$(size(itp,d))")
+    end
+
+    ScaledInterpolation(T,N,itp,IT,GT,ranges)
+end
+
+@generated function getindex{T,N,ITPT,IT<:DimSpec}(sitp::ScaledInterpolation{T,N,ITPT,IT}, xs...)
+    length(xs) == N || error("Must index into $N-dimensional scaled interpolation object with exactly $N indices (you used $(length(xs))")
+    if IT <: Tuple
+        # handle dimspecs
+        interp_indices = map(it -> IT.parameters[it] != NoInterp, 1:length(IT.parameters))
+        :(getindex(sitp.itp, map(coordlookup, $interp_indices, sitp.ranges, xs)...))
+    else
+        :(getindex(sitp.itp, coordlookup(sitp.ranges, tuple(xs...))...))
+    end
+end
+
+size(sitp::ScaledInterpolation, d) = size(sitp.itp, d)
+
+coordlookup(r::FloatRange, x) = (r.divisor * x - r.start) ./ r.step + one(eltype(r))
+coordlookup(r::StepRange, x) = (x - r.start) ./ r.step + one(eltype(r))
+coordlookup(r::UnitRange, x) = x - r.start + one(eltype(r))
+coordlookup(i::Bool, r::Range, x) = i ? coordlookup(r, x) : x
```

Comment:
I've changed this line locally to `coordlookup(i::Bool, r::Range, x) = i ? coordlookup(r,x) : convert(typeof(coordlookup(r,x)), x)` to try to fix the type instability (see comment below) but it doesn't help.